pac4j-cas/src/main/java/org/pac4j/cas/client/direct/DirectCasClient.java

```diff
@@ -0,0 +1,115 @@
+package org.pac4j.cas.client.direct;
+
+import org.jasig.cas.client.util.CommonUtils;
+import org.pac4j.cas.authorization.DefaultCasAuthorizationGenerator;
+import org.pac4j.cas.client.CasProxyReceptor;
+import org.pac4j.cas.config.CasConfiguration;
+import org.pac4j.cas.credentials.authenticator.CasAuthenticator;
+import org.pac4j.cas.profile.CasProfile;
+import org.pac4j.core.client.DirectClientV2;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.TokenCredentials;
+import org.pac4j.core.credentials.authenticator.Authenticator;
+import org.pac4j.core.credentials.extractor.ParameterExtractor;
+import org.pac4j.core.credentials.extractor.TokenCredentialsExtractor;
+import org.pac4j.core.exception.CredentialsException;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.http.CallbackUrlResolver;
+import org.pac4j.core.util.CommonHelper;
+
+/**
+ * <p>This class is the direct client to authenticate users on a CAS server for a web application in a stateless way: when trying to access a protected area,
+ * the user will be redirected to the CAS server for login and then back directly to this originally requested url.</p>
+ *
+ * <p>You should generally use the {@link org.pac4j.cas.client.CasClient} instead (this one is very specific and was designed for OAuth / OpenID Connect implementations in the CAS server.</p>
+ *
+ * <p>The configuration can be defined via the {@link #configuration} object.</p>
+ *
+ * <p>As no session is meant to be created, this client does not handle CAS logout requests.</p>
+ *
+ * <p>For proxy support, a {@link CasProxyReceptor} must be defined in the configuration (the corresponding "callback filter" must be enabled)
+ * and set to the CAS configuration of this client. In that case, a {@link org.pac4j.cas.profile.CasProxyProfile} will be return
+ * (instead of a {@link org.pac4j.cas.profile.CasProfile}) to be able to request proxy tickets.</p>
+ *
+ * @author Jerome Leleu
+ * @since 1.9.2
+ */
+public class DirectCasClient extends DirectClientV2<TokenCredentials, CasProfile> {
```

Comment:
@mmoayyed This is what you need to have a CAS stateless authentication -> https://github.com/apereo/cas/issues/1571